<?xml version="1.0"?>
<launch>

    <arg name="odom_topic" default="/odometry/filtered"/>   <!-- for evaluation -->

    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen" clear_params="true">

        <rosparam file="$(find kymco_maxxer90_navigation)/config/move_base_params.yaml" command="load" />

        <!-- Costmaps -->
        <rosparam file="$(find kymco_maxxer90_navigation)/config/costmap_common_params.yaml" command="load" ns="global_costmap" />
        <rosparam file="$(find kymco_maxxer90_navigation)/config/costmap_common_params.yaml" command="load" ns="local_costmap" />
        <rosparam file="$(find kymco_maxxer90_navigation)/config/local_costmap_params.yaml" command="load" />
        <rosparam file="$(find kymco_maxxer90_navigation)/config/global_costmap_params.yaml" command="load" />

        <!-- Local Planner -->
        <param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS" />
        <rosparam file="$(find kymco_maxxer90_navigation)/config/base_local_planner_params.yaml" command="load" />
        <param name="TebLocalPlannerROS/odom_topic" value="$(arg odom_topic)"/>     <!-- for evaluation -->

        <!-- Global Planner -->
        <param name="base_global_planner" value="SBPLLatticePlanner" />
        <rosparam file="$(find kymco_maxxer90_navigation)/config/sbpl_lattice_planner_params.yaml" command="load" />
        <!-- <param name="SBPLLatticePlanner/primitive_filename" value="$(find kymco_maxxer90_navigation)/motion_primitives/ackermann.mprim" /> -->
        <param name="SBPLLatticePlanner/primitive_filename" value="$(find kymco_maxxer90_navigation)/motion_primitives/ackermann_julia.mprim" />
        <!--<param name="SBPLLatticePlanner/primitive_filename" value="$(find kymco_maxxer90_navigation)/motion_primitives/alt/ackermann_julia.mprim" />-->

        <!-- CMD Topic -->
        <remap from="cmd_vel" to="kymco_maxxer90_ackermann_steering_controller/cmd_vel" />
    </node>
</launch>
